manager/manager.go

```diff
@@ -150,7 +150,7 @@ func NewManager(opts Options) (*Manager, error) {
 		lister:            lister,
 		metadataReader:    opts.MetadataReader,
 		clock:             opts.Clock,
-		log:               opts.Log,
+		log:               *opts.Log,
```

Comment:
This will cause a panic if a user doesn't provide a logger (whereas the previous code would not). Can we initialise a `logr.Discard()` here to pass to the Manager if it is nil?

Reply:
There's already a check for nil `opts.Log` above on line 102 https://github.com/cert-manager/csi-lib/pull/19/files#diff-f679b61276b44dc3f4712950536707c0389ee4c536fc29837c0202c6e81c1bbfR102